testsuite/python/reaction_methods.py

```diff
@@ -0,0 +1,241 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import espressomd
+import espressomd.reaction_ensemble
+
+import unittest as ut
+
+
+class ReactionMethods(ut.TestCase):
+
+    """Test the reaction methods interface."""
+
+    system = espressomd.System(box_l=[10., 10., 10.])
+    system.cell_system.skin = 0.4
+
+    def check_interface(self, method, kT, exclusion_radius, gamma):
+        def check_reaction_parameters(reactions, parameters):
+            for reaction, params in zip(reactions, parameters):
+                for key in reaction.required_keys():
+                    self.assertEqual(getattr(reaction, key), params[key])
+
+        reaction_forward = {
+            'gamma': gamma,
+            'reactant_types': [5],
+            'reactant_coefficients': [1],
+            'product_types': [2, 3],
+            'product_coefficients': [1, 1],
+            'default_charges': {5: 0, 2: 0, 3: 0},
+        }
+        reaction_backward = {
+            'gamma': 1. / gamma,
+            'reactant_types': reaction_forward['product_types'],
+            'reactant_coefficients': reaction_forward['product_coefficients'],
+            'product_types': reaction_forward['reactant_types'],
+            'product_coefficients': reaction_forward['reactant_coefficients'],
+            'default_charges': reaction_forward['default_charges'],
+        }
+
+        if isinstance(method, espressomd.reaction_ensemble.ConstantpHEnsemble):
+            method.add_reaction(gamma=reaction_forward['gamma'],
+                                reactant_types=reaction_forward['reactant_types'],
+                                product_types=reaction_forward['product_types'],
+                                default_charges=reaction_forward['default_charges'])
+        else:
+            method.add_reaction(**reaction_forward)
+        reaction_parameters = (reaction_forward, reaction_backward)
+
+        # check getters and setters
+        self.assertAlmostEqual(method.kT, kT, delta=1e-10)
+        self.assertAlmostEqual(
+            method.exclusion_radius,
+            exclusion_radius,
+            delta=1e-10)
+        self.assertAlmostEqual(
+            method.get_volume(),
+            self.system.volume(),
+            delta=1e-10)
+        method.set_volume(volume=1.)
+        self.assertAlmostEqual(method.get_volume(), 1., delta=1e-10)
+        self.assertEqual(method.get_non_interacting_type(), 100)
+        method.set_non_interacting_type(type=9)
+        self.assertEqual(method.get_non_interacting_type(), 9)
+        if isinstance(method, espressomd.reaction_ensemble.ConstantpHEnsemble):
+            self.assertAlmostEqual(method.constant_pH, -10., delta=1e-10)
```

Comment:
just out of interest, is there a reason to set the default pH value to `-10.0`? In an ideal world, shouldn't that be a parameter of the constructor?

Reply:
`-10` is the historical sentinel value for uninitialized class members in reaction method classes. In this specific case, I think there is no obstacle to making the constant pH a part of the core class constructor. @pm-blanco would you agree with that change? From the users perspective, it would still be possible to do a titration with `for pH in range(10): RE.constant_pH = pH`, but it would be mandatory to provide a sensible value in `RE = ConstantpHEnsemble(kT=1.5, exclusion_radius=0.8, seed=12, constant_pH=6)`.

If there is a constraint on the pH range, we should actually write a setter function in the core class that throws an exception if the pH is outside the valid range. I don't think that's the case though, because negative pH values are well-defined in non-aqueous solvents and obey the same thermodynamic equations.

Reply:
I agree with that change, I think that actually it is desired that one has to provide the pH when doing an instance of the constant pH ensemble, since one cannot use this method without setting the pH value. You are right @jngrad, it should not be any a priori restriction for the pH value.

Reply:
Done: `kT`, `exclusion_radius` and `constant_pH` are are now required arguments of the core class constructors.